src/transformState.js

```diff
@@ -6,6 +6,34 @@
  */
 function transformState(state, actions) {
   // write code here
+  for (let n = 0; n < actions.length; n++) {
+    switch (actions[n].type) {
+      case 'addProperties':
+        const actionEntries = Object.entries(actions[n].extraData);
+
+        for (const [key, value] of actionEntries) {
+          state[key] = value;
+        }
+
+        break;
+
+      case 'removeProperties':
+        for (let i = 0; i < actions[n].keysToRemove.length; i++) {
+          delete state[actions[n].keysToRemove[i]];
+        }
+
+        break;
+
+      case 'clear':
+        const stateKeys = [ ...Object.keys(state) ];
```

Comment:
Redundant variable creation: The stateKeys variable is not necessary, as you can simply iterate over the keys of the state object directly

Reply:
The variable is necessary because it is not a reference, it copies the keys table
When iterating over the keys of the state object with a for loop, it outputs the following error:

>   ● Should clear the state                                                                                                                                                                                                       
                                                                                                                                                                                                                                 
    expect(received).toEqual(expected) // deep equality

    - Expected
    + Received

    - Object {}
    + Object {
    +   "name": "Jim",
    + }

      161 |
      162 |   expect(state)
    > 163 |     .toEqual({});
          |      ^
      164 | });
      165 |
      166 | test('Should not fails when clearing an empty state', () => {

      at Object.<anonymous> (src/transformState.test.js:163:6)

Reply:
I will look into other ways of iterating through the object

Reply:
Worked perfectly fine while iterating from the end of the array instead; going to apply the approach